### PR TITLE
Add the option to recover a wallet to the Import Wallet start page

### PR DESCRIPTION
**Pull request by dauTT**
**What current issue(s) from Trello/Github does this address?**

#1621 

**What problem does this PR solve?**

This PR add the option to recover a wallet to the `Import Wallet` start page.

image

**How did you solve this problem?**
1) Refactor `CreateWallet.jsx`: move the `Form` component to a separate file, `CreateImportWalletForm.jsx` and split the logic of importing and creating wallet into two different containers (`CreateWallet`, `ImportWallet`) in the `CreateImportWallet` folder.
2) Extract `loadWalletRecovery` method from `Setting` class and make it exportable.
3) Add the option to recover a wallet in the `ImportWallet` container by reusing the `loadWalletRecovery` method.

**How did you make sure your solution works?**

Tested the code with different scenarios:
1) `Create Wallet`: with valid and invalid input parameters
2) `Import Wallet` with option`Private Key`, `Encrypted Key`, `Recover Wallet` and with valid and invalid input parameters
3) `Recover Wallet` in the `Setting` section.

**Are there any special changes in the code that we should be aware of?**

**Is there anything else we should know?**

- [ ] Unit tests written?

### CODE REVIEW

File: app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import styles from './ImportWallet.scss'
+
+type Props = {
+  // eslint-disable-next-line react/no-unused-prop-types
+  authenticated: boolean,
+  // eslint-disable-next-line react/no-unused-prop-types
+  setAccounts: (Array<Object>) => any,
+  // eslint-disable-next-line react/no-unused-prop-types
+  showSuccessNotification: Object => any,
+  // eslint-disable-next-line react/no-unused-prop-types
```
Comment (comountainclimber) [inline app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx]: Typically the pattern in the app has been to only declare props that are being used by the component but I dont mind this 👍

File: app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  showErrorNotification: Object => any,
```
Comment (comountainclimber) [inline app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx]: Can we update the type definition to be a little more specific here:  `({message: string}) => string`

File: app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+}
+
+type State = {
+  tabIndex: number,
+}
+
+const IMPORT_KEY_OPTIONS = {
+  WIF: {
+    render: (props: Props) => (
```
Comment (comountainclimber) [inline app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx]: I think we can get rid of the `// eslint-disable-next-line react/no-unused-prop-types` above if we just define the props inline here `(props: {authenticated: boolean})`

File: app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+import Button from '../../../components/Button'
+import CreateImportWalletForm from '../../../components/CreateImportWalletForm'
+import FullHeightPanel from '../../../components/Panel/FullHeightPanel'
+import { loadWalletRecovery } from '../../Settings/Settings'
+import ImportIcon from '../../../assets/icons/import.svg'
+import AddIcon from '../../../assets/icons/add.svg'
+import styles from './ImportWallet.scss'
+
+type Props = {
+  // eslint-disable-next-line react/no-unused-prop-types
```
Comment (comountainclimber) [inline app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx]: I think we can get rid of these by using maybe types here https://flow.org/en/docs/types/maybe/

File: app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+      <CreateImportWalletForm
+        option="IMPORT"
+        importKeyOption="ENCRYPTED_WIF"
+        authenticated={props.authenticated}
+      />
+    ),
+    display: 'Encrypted Key',
+  },
+
+  RECOVER_WALLET: {
```
Comment (comountainclimber) [inline app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx]: Can we add some instructions here to fill out the empty space?
```
  RECOVER_WALLET: {
    render: (props: Props) => (
      <Fragment>
        <p>
          Upload a JSON wallet recovery file here to add your accounts to Neon.
          This option is also available on the Settings page.
        </p>
        <div
          className={styles.buttonContainer}
          onClick={() => {
            loadWalletRecovery(
              props.showSuccessNotification,
              props.showErrorNotification,
              props.setAccounts,
            )
          }}
        >
          <Button
            renderIcon={AddIcon}
            type="submit"
            shouldCenterButtonLabelText
            primary
          >
            Import File
          </Button>
        </div>
      </Fragment>
    ),
    display: 'Recover Wallet',
  },```

Comment (comountainclimber) [reply app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx]: Screen Shot 2019-07-02 at 11 43 58 AM

File: app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx (hunk 4 above)
Comment (dauTT) [reply app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx]: If I add only one comment before the `Props` definition, it is still working:

```Javascript
// @flow
import React from 'react'
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs'
import { ROUTES } from '../../../core/constants'
import CloseButton from '../../../components/CloseButton'
import BackButton from '../../../components/BackButton'
import Button from '../../../components/Button'
import CreateImportWalletForm from '../../../components/CreateImportWalletForm'
import FullHeightPanel from '../../../components/Panel/FullHeightPanel'
import { loadWalletRecovery } from '../../Settings/Settings'
import ImportIcon from '../../../assets/icons/import.svg'
import AddIcon from '../../../assets/icons/add.svg'
import styles from './ImportWallet.scss'

/* eslint-disable react/no-unused-prop-types */
type Props = {
  authenticated: boolean,
  setAccounts: (Array<Object>) => any,
  showSuccessNotification: ({ message: string }) => string,
  showErrorNotification: ({ message: string }) => string,
}

```

**Are you ok, with this proposal?**

File: app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx (hunk 5 above)
Comment (dauTT) [reply app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx]: Nice suggestion! I will make the changes once we have clarified the point above related to the `eslint` warning.

File: app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx (hunk 4 above)
Comment (comountainclimber) [reply app/containers/CreateImportWallet/ImportWallet/ImportWallet.jsx]: This is fine @dauTT 👍

**Comment by comountainclimber** [comment]
@dauTT lets try and wrap up the small updates to your awesome work so we can get this into `dev`

**Comment by dauTT** [comment]
@comountainclimber ,

I believe that I have already addressed all the open items in my last commit. Let me know in case I am mistaken.

**Comment by comountainclimber** [comment]
👍 will take a look

**Comment by comountainclimber** [comment]
🎉  Please award @dauTT 500 points

**Comment by lllwvlvwlll** [comment]
https://neoscan.io/transaction/589328c622e101065f6717d61a332f93aac71cc68b665e194ef8c874a7e26669

**Pull request merged**